doc/userguide/rules/http-keywords.rst

```diff
@@ -1,813 +1,1200 @@
 
-http.request_header
--------------------
+.. _http.host:
 
-Match on the name and value of a HTTP request header (HTTP1 or HTTP2).
+http.host
+---------
 
-For HTTP2, name and value get concatenated by ": ", colon and space.
-To detect if a http2 header name contains ':',
-the keyword ``http2.header_name`` can be used.
+Matching on the HTTP host name has two options in Suricata, the ``http.host``
+and the ``http.host.raw`` sticky buffers.
 
-Examples::
+It is possible to use any of the :doc:`payload-keywords` with both ``http.host``
+keywords.
 
-  http.request_header; content:"agent: nghttp2";
-  http.request_header; content:"custom-header: I love::colons";
+.. note:: The ``http.host`` keyword normalizes the host header contents. If a
+  host name has uppercase characters, those would be changed to lowercase.
 
-``http.request_header`` is a 'sticky buffer'.
+Normalization Example::
 
-``http.request_header`` can be used as ``fast_pattern``.
+  GET /index.html HTTP/1.1
+  User-Agent: Mozilla/5.0
+  Host: SuRiCaTa.Io
 
+In the above example the host buffer would contain `suricata.io`.
 
-http.response_header
---------------------
+.. container:: example-rule
+
+  alert http $HOME_NET any -> $EXTERNAL_NET any (msg:"HTTP Host Example"; \
+  flow:established,to_server; :example-rule-options:`http.host; \
+  content:"suricata.io";` bsize:11; classtype:bad-unknown; sid:123; rev:1;)
+
+.. note:: The ``nocase`` keyword is no longer allowed since the host names
+  are normalized to contain only lowercase letters.
+
+.. note:: ``http.host`` does not contain the port associated with the host
+  (i.e. suricata.io:1234). To match on the host and port or negate a host
+  and port use :ref:`http.host.raw`.
+
+.. note:: ``http.host`` does not include the leading space or trailing \\r\\n
+
+.. note:: The ``http.host`` and ``http.host.raw`` buffers are populated
+  from either the URI (if the full URI is present in the request like
+  in a proxy request) or the HTTP Host header. If both are present, the
+  URI is used.
+
+.. note:: If a request contains multiple "Host" headers, the values will be
+  concatenated in the ``http.host`` and ``http.host.raw``
+  buffers, in the order seen from top to bottom, with a comma and space
+  (", ") between each of them.
+
+Example Duplicate Host Header Request::
+
+  GET /index.html HTTP/1.1
+  User-Agent: Chrome/2.0
+  Cookie: PHPSESSION=123
+  Host: suricata.io
+  Host: oisf.net
+
+.. container:: example-rule
+
+  alert http $HOME_NET any -> $EXTERNAL_NET any (msg:"HTTP Two Host Example"; \
+  flow:established,to_server; :example-rule-options:`http.host; \
+  content:"suricata.io, oisf.net";` classtype:bad-unknown; sid:125; rev:1;)
+
+.. _http.host.raw:
+
+http.host.raw
+-------------
+
+The ``http.host.raw`` buffer matches on HTTP host name content but does not
+have more than the standard normalization performed on the buffer contents.
+(see :ref:`http.host`)
```

Comment:
The way this is worded for `http.header.raw` is somewhat easier to follow and understand. Do they mean the same?